virttest/utils_test/libvirt.py

```diff
@@ -3550,3 +3550,25 @@ def check_qemu_cmd_line(content, err_ignore=False):
             raise exceptions.TestFail("Expected '%s' was not found in "
                                       "qemu command line" % content)
     return True
+
+
+def get_disk_alias(vm, source_file=None):
+    """
+    Get alias name of disk with given source file
+
+    :param vm: VM object
+    :param source_file: domain disk source file
+    """
+    if vm.is_alive():
+        ori_vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
+        disks = ori_vmxml.devices.by_device_tag('disk')
+        for disk in disks:
+            if disk.xmltreefile.find('source') is not None:
+                if disk.source.attrs['file'] == source_file:
+                    logging.info("Get alias name %s", disk.alias['name'])
+                    return disk.alias['name']
+            else:
+                if source_file is None:
+                    logging.info("Get alias %s for no source device", disk.alias)
+                    return disk.alias['name']
```

Comment:
if there are multiple disk, this line will return the first one. Is it acceptable ?

Reply:
If no source file, we can not judge which one is needed, and the current usage is for first, any more scenario?

Reply:
see above comments

Reply:
Assume below can satisfy your request:
```
for disk in disks:
  if disk.xmltreefile.find('source') is not None and disk.source.attrs['file'] != source_file:
          return None
  return disk.alias['name']
```

Reply:
Assume below code can satisfy your request:
```
def get_disk_alias(vm, source_file=None):
    """
    Get alias name of disk with given source file
    :param vm: VM object
    :param source_file: domain disk source file
    """
    disk_alias = None
    if vm.is_alive():
        ori_vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
        disks = ori_vmxml.devices.by_device_tag('disk')
        for disk in disks:
	    is_source_find = disk.xmltreefile.find('source')
	    if (not is_source_find and not source_file) or (is_source_find and disk.source.attrs['file'] == source_file)
		return disk.alias['name']
	else:
	    continue
return None
```

Reply:
Acceptable for me, will update soon, thanks